File: test/call_fortran.f90
!
! Call Fortran from Julia
!
module myfunctions
use, intrinsic :: iso_c_binding, only: c_double
implicit none
contains

function mul(x, y) result(r) bind(c)
    !GCC$ ATTRIBUTES DLLEXPORT :: mul
    real(kind=c_double), intent(in), value :: x, y
    real(kind=c_double) :: r
    r = x * y
end function

function sub(x, y) result(r) bind(c)
    !GCC$ ATTRIBUTES DLLEXPORT :: sub
    real(kind=c_double), intent(in), value :: x, y
    real(kind=c_double) :: r
    r = x - y
end function

end module myfunctions

program main
use julia
implicit none

! Load Julia into Fortran process
call julia_init()

! Run Julia program `call_fortran.jl`
! `call_fortran.jl` will call Fortran functions `mul` and `sub`
call julia_run("call_fortran.jl")

! Bye
call julia_destroy()
end program

File: call_fortran.jl
if Sys.iswindows()
# ccall Fortran function "mul(x, y)"
println(ccall(:mul, Float64, (Float64, Float64), 2.0, 3.0))

# ccall Fortran function "sub(x, y)"
println(ccall(:sub, Float64, (Float64, Float64), 10.0, 2.0))
end
